### DISCUSSION TITLE
Feature request: Add a snowflake extension

### DISCUSSION THREAD

**Discussion by lev112** [duckdb/duckdb#13328] [Ideas]
**Why do you want this feature?**
Snowflake is a very popular data warehouse, and while snowflake has support for iceberg tables,
it would be very useful to have access to internal snowflake tables from duckdb, 
for both querying and inserting data.

snowflake already has an [ODBC driver](https://docs.snowflake.com/en/developer-guide/odbc/odbc) available

**Comment by Hugoberry** [comment]
I've developed a cross-platform ODBC community extension: nanodbc. I've tested the Snowflake connectivity via ODBC on Linux and Windows. Here is an example of query I've tested with;
```sql
install nanodbc from community;
load nanodbc;

from odbc_query(
         connection='snow-second', --DSN with externalbrowser auth that supports SSO
         query='select * from sf_adventure.dbo.auditlog limit 20000000', 
         encoding='WINDOWS-1252');
```
